scripts/rosversion

```diff
@@ -68,47 +69,75 @@ def get_distro_name_from_roscore():
         traceback.print_exc()
 
 
-if len(sys.argv) == 2:
-    stack_name = sys.argv[1]
+def get_version_printer(skipnewline=False):
+    """
+    @return: A method object to print the version string.
+    """
+    if skipnewline:
+        return sys.stderr.write
+    else:
+        return print
 
-    if stack_name == '-d' or stack_name == '--distro':
-        if 'ROS_DISTRO' in os.environ:
-            distro_name = os.environ['ROS_DISTRO']
-        else:
-            distro_name = get_distro_name_from_roscore()
-        if not distro_name:
-            print('<unknown>')
-        else:
-            print(distro_name)
-        sys.exit(0)
+parser = argparse.ArgumentParser(description=USAGE)
+if len(sys.argv) < 2:
```

Comment:
This check is not valid. When the user e.g. passes `-s` the condition is `True` without having passed any of the required arguments.

I changed the patch to pass `required=True` to the `add_mutually_exclusive_group()` call.